### ISSUE TITLE
tests on develop are failing!

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#138]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/138) by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

Looks like cab05c7 is the first failing commit, so I think was caused by Dan. Any ideas Dan?

It looks like a very small edit, so we should at least check if it passes if we revert the edit.

**Comment by joefowler** [comment]
**Original comment by Dan Becker (Bitbucket: [danbek](https://bitbucket.org/danbek), GitHub: [danbek](https://github.com/danbek)).**

----------------------------------------

How can  you tell that tests are failing on develop? I just ran the tests on my local machine in develop (after pulling the latest changes), and everything was fine. I also don't see anything about failing tests on the main project page.

**Comment by joefowler** [comment]
**Original comment by Dan Becker (Bitbucket: [danbek](https://bitbucket.org/danbek), GitHub: [danbek](https://github.com/danbek)).**

----------------------------------------

I can reproduce a failing test on the beckerd branch. It was actually caused by commit 5604aa0.

This commit makes a small change to the logic used to decide whether a noise record had a pulse. The logics rejects a "chunk" if it has a value too far away from a mean value. I changed the calculation of the mean value to occur at the "chunk" level instead of the "segment" level. This works better for umux data where we have flux jumps (so that a jump causes only a single "chunk" to be thrown away, instead of an entire segment.

Because the mean is subtracted from each chunk prior to calculating the autocorrelation, this has the effect of making very small changes in the autocorrelation function (because each "chunk" in a segment has a different mean subtracted, instead of all "chunks" in a segment having the same mean subtracted). This then causes very small changes in the filter, which leads to very small difference in filtered pulse height (1 part in 10^6).

Since the changes in filtered value are so small, I'm going to change the expected values to be the same as what the new code is outputting.

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

How to tell the tests are failing on develop:
![joe_fowler___Mass___Commits_—_Bitbucket.png](https://bitbucket.org/repo/KrBpjo/images/3786399969-joe_fowler___Mass___Commits_%E2%80%94_Bitbucket.png)

**Comment by joefowler** [comment]
**Original comment by Dan Becker (Bitbucket: [danbek](https://bitbucket.org/danbek), GitHub: [danbek](https://github.com/danbek)).**

----------------------------------------

OK, I see that now. But when I use the develop branch on my own computer and use `python runtest.py`, no errors are reported. So how do I reproduce the failing tests? Are you able to reproduce them locally?

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

The tests pass locally on my machine. They only fail one one of the two test platforms as well, which is a bit odd.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

We need to fix this. But I don't understand how. Anyone? Should we (temporarily) disable a certain test?

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

I'm tempted, but I'm also worried temporary wouldn't be temporary. I'm going to try going back to a commit just before the failures started and see if the tests pass.

**Comment by joefowler** [comment]
**Original comment by Dan Becker (Bitbucket: [danbek](https://bitbucket.org/danbek), GitHub: [danbek](https://github.com/danbek)).**

----------------------------------------

The errors seem to be from test_tail:

https://bitbucket.org/joe_fowler/mass/addon/pipelines/home#!/results/208

I don't know why it isn't failing for us locally. Maybe something about the version of libraries available on the build server?

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

I went back to commit b5c7b17 (1 month ago) and made a new branch, and a commit, to make the automated tests run. The tests ran fine.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

I'm confused: where do we have a 100 tests per month limit? At semaphore? Or in Bitbucket pipelines?

Is this limit going to make it hard to track down the problem, as we are unable to reproduce it on our own machines?

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

semaphore has the limit, bitbucket appears to be unlimited as far as I can tell

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

I don't know why, but the tests passed in `quality_of_life_galen` branch (which I had merged develop into). So I merged it, and now the tests pass on develop. Confusing and annoying. Hopefully the requirement for pull requests will help us catch any new problems before they get onto the branch.
